yt_dlp/extractor/alphaporno.py

```diff
@@ -1,75 +1,142 @@
+                'thumbnail': r're:https?://.*\.jpg$',
+                'timestamp': 1418701811,
+                'upload_date': '20141216',
+                'duration': 387,
+                'categories': list,
+                'age_limit': 18,
+            },
+        },
+        {
+            'url': 'https://www.alphaporno.com/videos/amazing-inches-hammering-her-pussy-in-such-addictive-ways/',
+            'info_dict': {
+                'id': '433761',
+                'ext': 'mp4',
+                'title': 'Amazing inches hammering her pussy in such addictive ways',
+                'display_id': 'amazing-inches-hammering-her-pussy-in-such-addictive-ways',
+                'timestamp': 1641065820,
+                'upload_date': '20220101',
+                'description': 'md5:8bf8e04807b890b847cc9238c445783a',
+                'categories': 'count:12',
+                'age_limit': 18,
+                'thumbnail': r're:https?://.*\.jpg$',
+                'duration': 298.0,
+            },
         },
-    }
+        {
+            'url': 'https://www.alphaporno.com/videos/anal-threesome-for-girls-younger-than-the-average/',
+            'info_dict': {
+                'id': '435603',
+                'ext': 'mp4',
+                'duration': 358.0,
+                'description': 'md5:bf8ca502575c20e15f4f33740cd20a94',
+                'categories': 'count:21',
+                'title': 'Anal threesome for girls y***er than the average',
+                'display_id': 'anal-threesome-for-girls-younger-than-the-average',
+                'upload_date': '20220209',
+                'thumbnail': r're:https?://.*\.jpg$',
+                'age_limit': 18,
+                'timestamp': 1644387720,
+            },
+        },
+    ]
 
     def _real_extract(self, url):
         display_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, display_id)
+        webpage, urlh = self._download_webpage_handle(url, display_id)
 
+        info = {
+            'display_id': display_id,
+        }
         video_id = self._search_regex(
             r"video_id\s*:\s*'([^']+)'", webpage, 'video id', default=None)
+        if video_id:
+            info['url'] = self._search_regex(
+                r"video_url\s*:\s*'([^']+)'", webpage, 'video url')
+            info['ext'] = self._html_search_meta(
+                'encodingFormat', webpage, 'ext', default='.mp4')[1:]
+        else:
+            video_id = self._search_regex(
+                r'video_id=(\d+)\b', webpage, 'video id')
+            formats = []
+            joined_url = urljoin(urlh.url, rf'/get_file/\d.+?/{video_id}/{video_id}_(\w+)\..+?')
+            for video_url, res in re.findall(rf'''({joined_url})(?:'|"|\b)\s''', webpage):
+                fmt = {
+                    'format_id': f'f{res}',
+                    'url': video_url,
+                }
+                fmt.update(parse_resolution(res) or {})
```

Comment:
`parse_resolution` already defaults to `{}`
```suggestion
                fmt.update(parse_resolution(res))
```